### PR TITLE
Verify with RRSIG

### PR DESCRIPTION
**Pull request by khibino**
Add logic to verify RR with RRSIG and DNSKEY.

implemented pubkey algorithm list
- RSA/SHA1 RSA/SHA256 RSA/SHA512
- ECDSA/P256 ECDSA/P384
- Ed25519 Ed448

### CODE REVIEW

File: dnsext-dnssec/DNS/SEC/Verify/RSA.hs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+-- https://datatracker.ietf.org/doc/html/rfc8017#section-9.2
+
+     SHA-1:       (0x)30 21 30 09 06 05 2b 0e 03 02 1a 05 00 04 14 || H.
+     SHA-224:     (0x)30 2d 30 0d 06 09 60 86 48 01 65 03 04 02 04 05 00 04 1c || H.
+     SHA-256:     (0x)30 31 30 0d 06 09 60 86 48 01 65 03 04 02 01 05 00 04 20 || H.
+     SHA-384:     (0x)30 41 30 0d 06 09 60 86 48 01 65 03 04 02 02 05 00 04 30 || H.
+     SHA-512:     (0x)30 51 30 0d 06 09 60 86 48 01 65 03 04 02 03 05 00 04 40 || H.
+ -}
+sha1   :: (SHA1,   ByteString)
+sha1    = (SHA1,   BS.pack [0x30, 0x21, 0x30, 0x09, 0x06, 0x05, 0x2b, 0x0e, 0x03, 0x02, 0x1a, 0x05, 0x00, 0x04, 0x14])
```
Comment (kazu-yamamoto) [inline dnsext-dnssec/DNS/SEC/Verify/RSA.hs:89]: You can write "\x30\x21\x30...".

Comment (khibino) [reply dnsext-dnssec/DNS/SEC/Verify/RSA.hs]: https://github.com/kazu-yamamoto/dnsext/pull/21/commits/76f579ffd9dc6f5618330c1989661aa0b060a8ee

File: dnsext-dnssec/DNS/SEC/Verify/RSA.hs [hunk 2]
```diff
@@ -86,10 +86,10 @@ rsaVerify (alg, pkcs1) pubkey sig msg = do
      SHA-512:     (0x)30 51 30 0d 06 09 60 86 48 01 65 03 04 02 03 05 00 04 40 || H.
  -}
 sha1   :: (SHA1,   ByteString)
-sha1    = (SHA1,   BS.pack [0x30, 0x21, 0x30, 0x09, 0x06, 0x05, 0x2b, 0x0e, 0x03, 0x02, 0x1a, 0x05, 0x00, 0x04, 0x14])
+sha1    = (SHA1,   B8.pack "\x30\x21\x30\x09\x06\x05\x2b\x0e\x03\x02\x1a\x05\x00\x04\x14")
```
Comment (kazu-yamamoto) [inline dnsext-dnssec/DNS/SEC/Verify/RSA.hs:89]: `B8.pack` is not necessary!

Comment (khibino) [reply dnsext-dnssec/DNS/SEC/Verify/RSA.hs]: https://github.com/kazu-yamamoto/dnsext/pull/21/commits/05b9839a16f2ca714419533fe81aca2d06903427

**Review by kazu-yamamoto — APPROVED** [review]
LGTM

**Comment by kazu-yamamoto** [comment]
Merged.

**Pull request merged**